### PR TITLE
[NPU] support npu for memcpy op

### PR DESCRIPTION
**Pull request by zhiqiu**
### PR types

New features
### PR changes

OPs
### Describe

[NPU] support npu for memcpy op
image
image

### CODE REVIEW

File: python/paddle/fluid/tests/unittests/npu/test_memcpy_op_npu.py [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+                    "dtype": npu_var.dtype,
+                    "value": 1.0,
+                    "place_type": 1
+                })
+            main_program.global_block().append_op(
+                type="fill_constant",
+                outputs={"Out": cpu_var_name},
+                attrs={
+                    "shape": [10, 10],
+                    "dtype": npu_var.dtype,
```
Comment (pangyoki) [inline python/paddle/fluid/tests/unittests/npu/test_memcpy_op_npu.py]: not important. `cpu_var.dtype`

**Review by pangyoki — DISMISSED** [review]
LGTM

File: python/paddle/fluid/tests/unittests/npu/test_memcpy_op_npu.py [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+            attrs={'dst_place_type': 0})
+        place = fluid.NPUPlace(0)
+        exe = fluid.Executor(place)
+        npu_, cpu_ = exe.run(main_program,
+                             feed={},
+                             fetch_list=[npu_var.name, cpu_var.name])
+        self.assertTrue(np.allclose(npu_, cpu_))
+        self.assertTrue(np.allclose(cpu_, np.ones((10, 10))))
+
+    def test_cpu_cpoy_gpu(self):
```
Comment (liym27) [inline python/paddle/fluid/tests/unittests/npu/test_memcpy_op_npu.py]: test_cpu_cpoy_gpu -> test_cpu_cpoy_npu

**Review by liym27 — APPROVED** [review]
LGTM

**Pull request merged**